### PR TITLE
Add support for bitfields

### PR DESCRIPTION
**Pull request by tmm1**
will close https://github.com/nim-lang/c2nim/issues/49

I'm emitting a `bitsize` pragma as @Varriount suggested, but it's currently showing up after the type instead of as part of it.

``` nim
type
  bits* = object
    flag*: cint
    {.bitsize:1.}
    opts*: cint
    {.bitsize:4.}
```

### CODE REVIEW

File: cparse.nim [hunk 1]
```diff
@@ -776,6 +776,13 @@ proc parseStructBody(p: var Parser, stmtList: PNode, isUnion: bool,
       t = parseTypeSuffix(p, t)
       addSon(def, i, t, ast.emptyNode)
       addSon(result, def)
+      if p.tok.xkind == pxColon:
+        getTok(p)
+        var bits = p.tok.iNumber
+        eat(p, pxIntLit)
+        var pragma = newNodeP(nkPragma, p)
+        addSon(pragma, newIdentNodeP("bitsize:" & $bits, p))
+        addSon(result, pragma)
```
Comment (tmm1) [inline cparse.nim]: I tried to `addSon(i, pragma)` to attach the pragma to the field, but it didn't work.

**Comment by Varriount** [comment]
Just to prevent confusion, the "bitsize" pragma isn't actually a thing yet. I just pulled it out of the air as a way one might mark type members as having a certain size.

**Comment by tmm1** [comment]
Maybe it's simpler to use array[0..x, char] for now?

**Comment by Varriount** [comment]
But a char is a byte long. Don't bitfields allow for members less than than sizeof(char)?

**Comment by tmm1** [comment]
Oops, you're right. I got my bits and bytes mixed up. Adding this pragma makes sense to me. I'm going to take a stab at implementing it in `nim` unless someone comes up with a better solution.

**Comment by Varriount** [comment]
The other alternative is to manually extract the data using a member access procedure (`proc \`.\`(t: expr): expr` if I recall correctly) and bit manipulation.

**Pull request merged**